packages/builder/src/pages/builder/app/[application]/design/[screenId]/components/[componentId]/_components/settings/ComponentSettingsPanel.svelte

```diff
@@ -25,32 +27,75 @@
   )
   $: isScreen = $selectedComponent?._id === $selectedScreen?.props._id
   $: title = isScreen ? "Screen" : $selectedComponent?._instanceName
+
+  let section = "general"
+  let componentInstanceId = null
+
+  const tabs = ["general", "styles", "conditions"]
+  const updateSection = () => {
+    if (componentInstanceId !== $selectedComponent?._id) {
+      section = tabs[0]
+      componentInstanceId = $selectedComponent?._id
+    }
+  }
+  $: updateSection(componentInstance)
```

Comment:
NAB, but a nice refactor of the check to reset the section every time the component changes would be:
```js
$: id = $selectedComponent?._id
$: id, (section = tabs[0])
```
The first line is required to ensure that it memoizes a primitive to avoid firing the reactive statement every time any property inside the selected component property changes.

Reply:
Nice, this is much neater!